### ISSUE TITLE
Questions about VectorLoad in LLVM IR dumped by XLA

### ISSUE THREAD

**Issue by Tengfei09** [openxla/xla#3587]
Hi, recently, I'm trying to analyze the performance of softmax with openXLA. Here is my simple test code.

```
from jax import numpy as jnp
import numpy as np
import jax

@jax.jit
def softmax(x):
    return jax.nn.softmax(x * 0.01)

arr = np.random.randn(16,4096,4096).astype(jnp.float16)

jax.profiler.start_trace("./log")
for i in range(100):
    softmax(arr)
jax.profiler.stop_trace()
```
After enabling XLA and setting the `export XLA_FLAGS="--xla_dump_to=./output"`, I got two different LLVM IRs, one is _ir-no-opt.ll,_ another is _ir-with-opt.ll_. 

When comparing two LLVM IRs, I found that certain LLVM passes have converted scalar load to vector load as shown below.

**ir-no-opt.ll**
```
%37 = getelementptr inbounds half, ptr %arg2, i32 %linear_index1
  %38 = load half, ptr %37, align 2, !invariant.load !11
```
**ir-with-opt.ll**
```
 %arg37 = addrspacecast ptr %arg3 to ptr addrspace(1)
  %arg25 = addrspacecast ptr %arg2 to ptr addrspace(1)
  %arg13 = addrspacecast ptr %arg1 to ptr addrspace(1)
  %arg01 = addrspacecast ptr %arg0 to ptr addrspace(1)
  %0 = tail call i32 @llvm.nvvm.read.ptx.sreg.ctaid.x(), !range !8
  %1 = shl nuw nsw i32 %0, 10
  %2 = tail call i32 @llvm.nvvm.read.ptx.sreg.tid.x(), !range !9
  %3 = shl nuw nsw i32 %2, 2
  %linear_index_base = or i32 %1, %3
  %4 = lshr i32 %0, 2
  %5 = zext i32 %linear_index_base to i64
  %6 = getelementptr half, ptr addrspace(1) %arg25, i64 %5
  %7 = load <4 x half>, ptr addrspace(1) %6, align 8, !invariant.load !10
  %8 = extractelement <4 x half> %7, i32 0
  %9 = extractelement <4 x half> %7, i32 1
  %10 = extractelement <4 x half> %7, i32 2
  %11 = extractelement <4 x half> %7, i32 3
```

**Question:** Which LLVM passes did this work?

by the way, I have tried to dump all LLVM passes by setting `export XLA_FLAGS="--xla_gpu_dump_llvmir" `， but didn't find any useful cues. 

Any cues or ideas would be highly appreciated. Thanks.
